ouroboros-consensus/src/Ouroboros/Consensus/Storage/ChainDB/Impl/VolDB.hs

```diff
@@ -497,26 +503,36 @@ blockFileParser VolDbArgs{..} =
     blockFileParser'
       volHasFS
       volGetBinaryBlockInfo
-      (decodeDisk volCodecConfig)
+      decodeNestedCtxtAndBlock
       volCheckIntegrity
       volValidation
+  where
+    prefixLen :: PrefixLen
+    prefixLen = reconstructPrefixLen (Proxy @(Header blk))
+
+    -- Also extract the prefix from the given bytestring
+    decodeNestedCtxtAndBlock
```

Comment:
This requires a comment if nothing else. I guess here we are in control of that bytestring, and so I _guess_ it really must _just_ be the block? In general though this is certainly not the case for those `ByteString -> ..` decoders, which get their compositionality from the `ByteSpan`s in the decoded value, right?